### PR TITLE
Implement solved challenge tracking across page loads.

### LINKED ISSUE #39: Track solved challenges for users
It would be cool if we persisted solved challenges for the user (perhaps highlighting them in green once they've been solved, for instance).

We could do this as simply as using a cookie/local storage, or using some sort of session variable.

As a stretch goal, I think it would be cool to store "who" solved which challenges on our side somewhere, which would let us see how many people are really solving each set of challenges each week/amount of interest different challenges are generating. Identity information should obviously be anonymized, if we end up going this route.

### PR DESCRIPTION
**Pull request by quixotrykd**
This commit also refactors flag checking functionality to better
document existing functions.

This commit closes #39.

### CODE REVIEW

**Comment by quixotrykd** [comment]
This commit should make relatively minimal changes to the look/feel of the page, but I have admittedly minimal experience with CSS/SCSS, so if there's a better way to do what I've done here feel free to fix it and/or let me know and I'll fix it.

File: assets/js/checkFlag.js [hunk 1]
```diff
@@ -1,12 +1,89 @@
...
+        let flagMd5 = hex_md5(flag);
+        
+        if(flagMd5.toLowerCase() === correctMd5.toLowerCase()) {
+            challengeForm.classList.remove("challenge-incorrect");
+            challengeForm.classList.add("challenge-solved");
+            storeChallengeSolved(challName);
+            return true;
+        }
+
+        challengeForm.classList.remove("challenge-solved");
```
Comment (napkinbrown) [inline assets/js/checkFlag.js:36]: "Oh you solved this challenge? Well not anymore"

Comment (quixotrykd) [reply assets/js/checkFlag.js]: I figured people may want to remove the solved status temporarily, if they want to check the flag again for whatever reason. The solve status will come back on page reload (i.e., incorrectly answering a challenge after it's been solved doesn't permanently remove the solved status).

**Comment by quixotrykd** [comment]
### No entry
image

### Incorrect flag
image

### Solved solution, just submitted
image

### Solved challenge, page load
image

### Incorrect answer, after previous solve
image

**Review by napkinbrown — APPROVED** [review]

**Pull request merged**